overlord/servicestate/quota_handlers.go

```diff
@@ -126,6 +126,14 @@ func quotaCreate(st *state.State, t *state.Task, action QuotaControlAction, allG
 		return fmt.Errorf("internal error, MemoryLimit option is mandatory for create action")
 	}
 
+	// make sure the memory limit is at least 4K, that is the minimum size
+	// to allow nesting, otherwise groups with less than 4K will trigger the
+	// oom killer to be invoked when a new group is added as a sub-group to the
+	// larger group.
+	if action.MemoryLimit <= 4*quantity.SizeKiB {
+		return fmt.Errorf("memory limit for group %q is too small, 4KB is minimum size", action.QuotaName)
```

Comment:
minimum size is actually 4k+1  ?  that is a bit confusing

Reply:
what I mean is that the error is ambiguous. It should say something like:

"size must be larger than 4KB" or "size must be >4KB"

Reply:
oh I see, yes that's a fair mount, let me adjust the error message then

Reply:
done